packages/core/test/trader/uniswap-broker/UniswapBroker.js

```diff
@@ -221,14 +219,12 @@ contract("UniswapBroker", function(accounts) {
 
     // Now we can actually execute the swapToPrice method to ensure that the contract correctly modifies the spot price.
     await uniswapBroker.swapToPrice(
+      true, // The swap is being executed as an EOA. This ensures that the correct token transfers are done.
       router.address,
       factory.address,
-      tokenA.address,
-      tokenB.address,
-      "1000", // The "true" price of the pair is expressed as the ratio of token A to token B. A price of 1000 is simply 1000/1.
-      "1",
-      MAX_UINT_VAL, // Set to the max posable value as we want to let the broker trade as much as needed in this example.
-      MAX_UINT_VAL,
+      [tokenA.address, tokenB.address],
+      ["1000", "1"], // The "true" price of the pair is expressed as the ratio of token A to token B. A price of 1000 is simply 1000/1.
```

Comment:
Unrelated to this line (and definitely not something we should do in this PR), but how hard do you think it would be to deploy a uniswap pair for this test and using it to vet the calculation?

Reply:
Take a look at the UniswapBroker tests in core ([here](https://github.com/UMAprotocol/protocol/blob/master/packages/core/test/trader/uniswap-broker/UniswapBroker.js)). They deploy a full uniswap factory & pair and goes through the motions of validating this behaviour. Is this sufficient?

Reply:
these tests _are_ using a real thing though. The `RangeTrader` tests deploy a `uniswapFactory` and create a `pair` from it. This dev is then seeded with an amount of tokens to create the initial price. The tests also use the normal `UniswapPriceFeed`, pointed at this "real" uniswap contract.